File: backend/Interfaces/Dtos/ProductDto.cs
using webapp_backend_dotnet.Model.Entities;

namespace webapp_backend_dotnet.Interfaces.Dtos
{
    public class ProductDto : ProductBaseDto
    {
        public string Number { get; }
        public int? SellingPrice { get; }
        public SupplierDto Supplier { get; }

        public ProductDto(Product product) : base(product)
        {
            if (product != null)
            {
                Number = product.Number;
                SellingPrice = product.SellingPrice;
                Supplier = new SupplierDto(product.Supplier);
            }
        }
    }
}

File: frontend/src/areas/products/types/productDto.ts
import { SupplierDto } from '../../suppliers/types/supplierDto';

export interface ProductDto {
    amountFactor: number;
    amountUnit: string;
    id: number;
    name: string;
    number: string;
    purchasingPrice: number;
    sellingPrice: number;
    supplier: SupplierDto;
}
